File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> a;
        int n=nums.size();
        
        for(int i=0;i<n;i++){
            int comp=target-nums[i];
            if(a.count(comp)){
                return {a[comp],i};
            }
            a[nums[i]]=i;
        }
        return {};
    }
};

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        nummap = {}  # Dictionary to store value -> index
        n = len(nums)

        for i in range(n):
            comp = target - nums[i]
            if comp in nummap:
                return [nummap[comp], i]
            nummap[nums[i]] = i

        return []